app/models/challenge_phases_evaluator.rb

```diff
@@ -15,4 +15,14 @@ class ChallengePhasesEvaluator < ApplicationRecord
   belongs_to :challenge
   belongs_to :phase
   belongs_to :user
+
+  validate :user_has_valid_role, if: -> { user.present? }
```

Comment:
`challenge_id` `phase_id` and `user_id` are all required (`NOT NULL`) fields. Rails also validates the presence of `belongs_to` relationships (unless you declare them with `optional: true`). so basically, `user.present?` check is not needed.